frontend/src/util/auth.tsx

```diff
@@ -77,9 +92,119 @@ class SimpleAuth {
 	}
 }
 
+class PasswordAuth {
+	currentUser: User | null = null
+	googleProvider?: Firebase.auth.GoogleAuthProvider
+	githubProvider?: Firebase.auth.GithubAuthProvider
+
+	constructor() {
+		this.initialize()
+	}
+
+	initialize() {
+		const user = sessionStorage.getItem('currentUser')
+		const token = sessionStorage.getItem('passwordToken')
+		if (user && token) {
+			try {
+				const userObject = JSON.parse(user)
+				this.currentUser = makePasswordAuthUser(userObject.email)
+			} catch (error) {
+				console.log('error setting user from session storage')
+			}
+			this.validateTokenAndInitializeUser(token)
+		}
+	}
+
+	async validateTokenAndInitializeUser(token: string): Promise<void> {
```

Comment:
This is probably redundant but seemed called for given my attempts
to tighten the password-auth implementation